gameplay/avalonRoom.js

```diff
@@ -1388,11 +1389,24 @@ module.exports = function (host_, roomId_, io_, maxNumPlayers_, newRoomPassword_
 			}
 		}
 
-
 		// If they have not been kicked before
-		if (thisRoom.kickedPlayers[socket.request.user.username] === true) {
-			socket.emit("danger-alert", "You have been banned from this room. You can not join.");
-			return false;
+		if (Object.keys(thisRoom.kickedPlayers).includes(socket.request.user.username) == true){
+			if(thisRoom.kickedPlayers[socket.request.user.username]["status"] === true) {
+				var banTimeLeft = banTime - ((new Date).getTime() - thisRoom.kickedPlayers[socket.request.user.username]["startTime"]);
+				if (banTimeLeft > 0) {
+					var banMsg = "You have been banned from joining the game for the next " + 
+						convertMillisToSeconds(banTimeLeft) + ". You can not join yet.";
+					socket.emit("danger-alert", banMsg);
+					return false;
+				} else {
+					//if ban time has been depleted, allow player to join the room again
+					//by removing him from the kickedPlayers map
+					delete thisRoom.kickedPlayers[socket.request.user.username];
+					console.log(socket.request.user.username + " has been removed from kickedPlayer list of room # "+ thisRoom.roomId);
+					socket.emit("danger-alert", null, "hide");
```

Comment:
Do we need this?

Reply:
Good to have but can be removed.
Will separate non-topics next time.
Accepted. Will be addressed in the next commit.

Reply:
Yes. I understand. But I just noticed that I'm not addressing the bug itself in its context and I often introduce new features that are optional within a bugfix. An old habit that has to die. :) not a good practice.